crates/optimism/cli/src/commands/init_state/bedrock.rs

```diff
@@ -0,0 +1,132 @@
+use reth_primitives::{
+    BlockBody, Header, SealedBlock, SealedBlockWithSenders, SealedHeader, StaticFileSegment, U256,
+};
+use reth_provider::{
+    providers::StaticFileProvider, BlockWriter, DatabaseProviderRW, StageCheckpointWriter,
+    StaticFileWriter,
+};
+use reth_stages::{StageCheckpoint, StageId};
+use tracing::info;
+
+/// Creates a dummy chain (with no transactions) up to the last OVM block and appends the
+/// first valid Bedrock block.
+pub(crate) fn setup_op_mainnet_without_ovm<DB: Database>(
+    provider_rw: &DatabaseProviderRW<DB>,
+    static_file_provider: &StaticFileProvider,
+) -> Result<(), eyre::Error> {
+    info!(target: "reth::cli", "Setting up dummy OVM chain before importing state.");
+
+    // Write OVM dummy data up to `BEDROCK_HEADER - 1` block
+    append_dummy_chain(static_file_provider)?;
+
+    info!(target: "reth::cli", "Appending Bedrock block.");
+
+    append_bedrock_block(provider_rw, static_file_provider)?;
+
+    for stage in StageId::ALL {
+        provider_rw.save_stage_checkpoint(stage, StageCheckpoint::new(BEDROCK_HEADER.number))?;
+    }
+
+    info!(target: "reth::cli", "Set up finished.");
+
+    Ok(())
+}
+
+/// Appends the first bedrock block.
+///
+/// By appending it, static file writer also verifies that all segments are at the same
+/// height.
+fn append_bedrock_block<DB: Database>(
+    provider_rw: &DatabaseProviderRW<DB>,
+    sf_provider: &StaticFileProvider,
+) -> Result<(), eyre::Error> {
+    provider_rw.insert_block(
+        SealedBlockWithSenders::new(
+            SealedBlock::new(
+                SealedHeader::new(BEDROCK_HEADER, BEDROCK_HEADER_HASH),
+                BlockBody::default(),
+            ),
+            vec![],
+        )
+        .expect("no senders or txes"),
+    )?;
+
+    sf_provider.latest_writer(StaticFileSegment::Headers)?.append_header(
+        &BEDROCK_HEADER,
+        BEDROCK_HEADER_TTD,
+        &BEDROCK_HEADER_HASH,
+    )?;
+
+    sf_provider
+        .latest_writer(StaticFileSegment::Receipts)?
+        .increment_block(BEDROCK_HEADER.number)?;
+
+    sf_provider
+        .latest_writer(StaticFileSegment::Transactions)?
+        .increment_block(BEDROCK_HEADER.number)?;
+
+    Ok(())
+}
+
+/// Creates a dummy chain with no transactions/receipts up to `BEDROCK_HEADER - 1` block.
+///
+/// * Headers: It will push an empty block.
+/// * Transactions: It will not push any tx, only increments the end block range.
+/// * Receipts: It will not push any receipt, only increments the end block range.
+fn append_dummy_chain(sf_provider: &StaticFileProvider) -> Result<(), eyre::Error> {
+    let mut headers_writer = sf_provider.latest_writer(StaticFileSegment::Headers)?;
+    let txs_writer = sf_provider.latest_writer(StaticFileSegment::Transactions)?;
+    let receipts_writer = sf_provider.latest_writer(StaticFileSegment::Receipts)?;
+    let (tx, rx) = std::sync::mpsc::channel();
+
+    rayon::scope(|s| {
```

Comment:
this should be fine because this is the only thing we're running, even if this does a lot of IO.

still would prefer regular thread for this.